packages/test/test-end-to-end-tests/src/test/summaryDataStoreStats.spec.ts

```diff
@@ -87,9 +87,17 @@ describeNoCompat("Generate Summary Stats", (getTestObjectProvider) => {
      * @returns the sequence number of the summary
      */
      async function waitForSummary(): Promise<number> {
-        await provider.ensureSynchronized();
+        // create the timeout error message since the timeout reason in local test is still not clear
+        // (Bug 1556 on sprint board)
+        await timeoutAwait(provider.ensureSynchronized(), {
+            durationMs: 2000,
```

Comment:
i don't think we should use a constant timeout here. i would update the method to take a timeout param, and then in the test call `this.timeout()` which will return the current test timeout. i would then divide that number by 2 or 3, and use that as the timeout param passed here, as we want this timeout to always occur before the mocha timeout, so each timeoutAwait should only get a faction of the test timeout, and we want to leave some additional time for the rest of the test as well

Reply:
in the current code if the test timeout is 2seconds, the first await could run from 1999 ms, and then the test would still timeout and we would loose the callstack

Reply:
use the timeout param and fraction instead